WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/promptslist/compose/BloggingPromptsListItem.kt

```diff
@@ -0,0 +1,118 @@
+package org.wordpress.android.ui.bloggingprompts.promptslist.compose
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.sp
+import org.wordpress.android.R
+import org.wordpress.android.ui.bloggingprompts.promptslist.model.BloggingPromptsListItemModel
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.compose.unit.FontSize
+import org.wordpress.android.ui.compose.unit.Margin
+import org.wordpress.android.ui.compose.utils.asString
+import org.wordpress.android.ui.utils.UiString.UiStringPluralRes
+import org.wordpress.android.util.LocaleManager
+import java.text.SimpleDateFormat
+
+private const val MONTH_DAY_FORMAT = "MMM d"
+private val answeredColor = Color(0xFF008A20)
+
+private val ItemSubtitleTextStyle
+    @ReadOnlyComposable
+    @Composable
+    get() = TextStyle(
+            color = LocalContentColor.current.copy(alpha = 0.6f),
+            fontSize = FontSize.Small.value,
+            fontWeight = FontWeight.W400,
+            letterSpacing = 0.4.sp,
+    )
+
+@Composable
+fun BloggingPromptsListItem(
+    model: BloggingPromptsListItemModel,
+    modifier: Modifier = Modifier,
+) {
+    val context = LocalContext.current
+    val dateFormat = remember(context) {
+        SimpleDateFormat(MONTH_DAY_FORMAT, LocaleManager.getSafeLocale(context))
+    }
```

Comment:
@RenanLukas, I would also like your opinion and thoughts on this specific part.

I was in doubt if this date formatting should be part of the presentation layer or if the `BloggingPromptsListItemModel` should already have the `String` ready for display (meaning this formatting would need to happen in the UseCase, probably).

Also, having the "formatter" instantiated here will make each item in the list have its own instance of it, even though all of them could be the same. I thought about receiving that formatter as an argument, which would hoist its initialization to the `Screen` level, but then this individual component gets a bit harder to Preview (as it will depend on an object that depends on the context and affects the UI directly) and definitely harder to test (which we are not doing now, anyway).

wdyt?

Reply:
Hey @thomashorta  👋 

>if this date formatting should be part of the presentation layer or if the BloggingPromptsListItemModel should already have the String ready for display (meaning this formatting would need to happen in the UseCase, probably).
>Also, having the "formatter" instantiated here will make each item in the list have its own instance of it, even though all of them could be the same. I thought about receiving that formatter as an argument, which would hoist its initialization to the Screen level, but then this individual component gets a bit harder to Preview (as it will depend on an object that depends on the context and affects the UI directly) and definitely harder to test (which we are not doing now, anyway).

I usually like to let the View just display the received content (e.g. getting a string instead of directly formatting a `Date` object) for testability, but if we just move this to the `UseCase` level we might harm the reusability considering another screen might require a different date format. IMHO the best place to format the `Date` would be the `ViewModel`, but I think it would be overkill to map the `UseCase` returned object to another object mapped by the `ViewModel`, with all fields being the same except for the formatted `Date`.
How about getting the format (like `MONTH_DAY_FORMAT`) as parameter in the `UseCase`? WDYT? 🤔
```
suspend fun execute(dateFormat: String): Result

//or with a sealed class to make sure the format is valid

suspend fun execute(dateFormat: DateFormatSealedClass): Result
```

If the suggestion above makes sense, maybe we could inject `LocaleManagerWrapper` in the `UseCase` so the static call doesn't harm the testability.

Reply:
As we talked in DMs I went forward moving this to a `Mapper` class and using the `LocaleManagerWrapper`. So the mapper is injected in the `ViewModel` and Domain -> UI mapping happens there at the presentation layer but not in the Compose / UI scope, which is better for testability, reusability, and performance. Thanks for the review and suggestions!